grails-app/services/de/iteratec/osm/measurement/environment/wptserver/ResultPersisterService.groovy

```diff
@@ -233,48 +237,67 @@ class ResultPersisterService implements iResultListener {
     }
 
     @Transactional(propagation = Propagation.REQUIRES_NEW)
-    protected List<EventResult> persistResultsOfOneTeststep(Integer testStepZeroBasedIndex, WptResultXml resultXml) throws OsmResultPersistanceException {
-
+    protected boolean persistResultsOfOneTeststep(Integer testStepZeroBasedIndex, WptResultXml resultXml) throws OsmResultPersistanceException {
+        List<EventResult> resultsOfTeststep = []
         String testId = resultXml.getTestId()
-        String labelInXml = resultXml.getLabel()
-        JobResult jobResult = JobResult.findByJobConfigLabelAndTestId(labelInXml, testId)
-        if (jobResult == null) {
-            throw new OsmResultPersistanceException(
-                    "JobResult couldn't be read from db while persisting associated EventResults for test id '${testId}'!"
-            )
-        }
-        Job job = jobDaoService.getJob(labelInXml)
-        if (job == null) {
-            throw new OsmResultPersistanceException(
-                    "No Job exists with label '${labelInXml}' while persting associated EventResults!"
-            )
-        }
-
-        log.debug('getting event name from xml result ...')
-        String measuredEventName = resultXml.getEventName(job, testStepZeroBasedIndex)
-        log.debug('getting event name from xml result ... DONE')
-        log.debug("getting MeasuredEvent from eventname '${measuredEventName}' ...")
-        MeasuredEvent event = getMeasuredEvent(measuredEventName);
-        log.debug("getting MeasuredEvent from eventname '${measuredEventName}' ... DONE")
 
-        log.debug("persisting result for step=${event}")
-        Integer runCount = resultXml.getRunCount()
-        log.debug("runCount=${runCount}")
-
-        List<EventResult> resultsOfTeststep = []
-        resultXml.getRunCount().times { Integer runNumber ->
-            if (resultXml.resultExistForRunAndView(runNumber, CachedView.UNCACHED) &&
-                    (job.persistNonMedianResults || resultXml.isMedian(runNumber, CachedView.UNCACHED, testStepZeroBasedIndex))) {
-                EventResult firstViewOfTeststep = persistSingleResult(resultXml, runNumber, CachedView.UNCACHED, testStepZeroBasedIndex, jobResult, event)
-                if (firstViewOfTeststep != null) resultsOfTeststep.add(firstViewOfTeststep)
+        if (isEventResultValid(resultXml, testStepZeroBasedIndex)) {
```

Comment:
Could you invert the logic here?
If the eventResult is _not_ valid, log it and exit with `false`, otherwise just continue?
This way you don't need to indent the whole actual logic which makes the code more readable and the diff smaller.